integrations/launchdarkly/client.py

```diff
@@ -135,6 +139,43 @@ async def fetch_environments_for_project(
             environments.extend(updated_batch)
         return environments
 
+    async def get_feature_flag_status(
+        self, projectKey: str, featureFlagKey: str
+    ) -> dict[str, Any]:
+        endpoint = f"flag-status/{projectKey}/{featureFlagKey}"
+        feature_flag_status = await self.send_api_request(endpoint)
+        return feature_flag_status
+
+    async def get_paginated_feature_flag_statuses(
+        self,
+    ) -> AsyncGenerator[list[dict[str, Any]], None]:
+        async for environments in self.get_paginated_environments():
+            tasks = [
+                self.fetch_statuses_from_environment(environment)
+                for environment in environments
+            ]
+            for completed_task in asyncio.as_completed(tasks):
+                batch_statuses = await completed_task
+                yield batch_statuses
+
+    async def fetch_statuses_from_environment(
+        self, environment: dict[str, Any]
+    ) -> list[dict[str, Any]]:
+        resource = f"{environment['__projectKey']}/{environment['key']}"
+        all_statuses = []
+        async for statuses in self.get_paginated_resource(
+            kind=ObjectKind.FEATURE_FLAG_STATUS, resource_path=resource
+        ):
+            updated_batch = [
+                {
+                    **status,
+                    "__environmentKey": environment["key"],
+                }
+                for status in statuses
+            ]
+            all_statuses.extend(updated_batch)
+        return updated_batch

```

Comment:
Can you please use the merge stream utility function that @Tankilevitch added instead of waiting all the flags in an environment before returning

Reply:
@yairsimantov20, I'll glad if you could provide a reference to the function

Reply:
Fetch statuses should yield instead waiting for everything to return 

And instead of as completed use something like this

```
 async for resource_groups_batch in stream_async_iterators_tasks(tasks):
                    yield resource_groups_batch
```

Reply:
`fetch_statuses_from_environment`  returns a list instead of using yield because it was used inside asyncio.as_completed